### PR TITLE
fixing php 32 bit (arm) filemtime on large file issue (#18971)

### PR DESCRIPTION
**Pull request by cyberb**
Filemtime fix for 32bit platform only.
This contribution is MIT licensed

### CODE REVIEW

File: lib/private/Files/Storage/Local.php [hunk 1]
```diff
@@ -167,8 +167,15 @@ public function file_exists($path) {
 	}
 
 	public function filemtime($path) {
-		clearstatcache($this->getSourcePath($path));
-		return $this->file_exists($path) ? filemtime($this->getSourcePath($path)) : false;
+		$fullPath = $this->getSourcePath($path);
+		clearstatcache($fullPath);
+		if (!$this->file_exists($path)) {
```
Comment (icewind1991) [inline lib/private/Files/Storage/Local.php]: This could probably be inside the 32bit block to save a bit of time

Comment (cyberb) [inline lib/private/Files/Storage/Local.php]: Sorry not sure I get that, are you talking about moving file existence check inside 32bjt block?
I thought this is platform independent check which prevents us from doing any modify time work if file does not exist.

**Comment by cyberb** [comment]
Any updates?

File: lib/private/Files/Storage/Local.php [hunk 2]
```diff
@@ -167,8 +167,15 @@ public function file_exists($path) {
...
+			return false;
+		}
+		if (PHP_INT_SIZE === 4) {
+			return exec ('stat -c %Y '. escapeshellarg ($fullPath));
```
Comment (VicDeo) [inline lib/private/Files/Storage/Local.php]: @cyberb cast to int?

**Comment by cyberb** [comment]
this contribution is MIT licensed

**Comment by VicDeo** [comment]
:+1: Looks good

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**